lib/livebook_cli/deploy.ex

```diff
@@ -0,0 +1,221 @@
+    }
+  end
+
+  defp ensure_config!(config) do
+    log_debug("Validating config from options...")
+
+    errors =
+      Enum.reduce(config, %{}, fn
+        {key, value}, acc when value in ["", nil] ->
+          add_error(acc, normalize_key(key), "can't be blank")
+
+        {:session_token, value}, acc ->
+          if not String.starts_with?(value, @deploy_key_prefix) do
+            add_error(acc, normalize_key(:session_token), "must be a Livebook Teams Deploy Key")
+          else
+            acc
+          end
+
+        {:teams_key, value}, acc ->
+          if not String.starts_with?(value, @teams_key_prefix) do
+            add_error(acc, normalize_key(:teams_key), "must be a Livebook Teams Key")
+          else
+            acc
+          end
+
+        {:path, value}, acc ->
+          if not File.exists?(value) do
+            add_error(acc, normalize_key(:path), "must be a valid path")
+          else
+            acc
+          end
+
+        _otherwise, acc ->
+          acc
+      end)
+
+    if Map.keys(errors) == [] do
+      :ok
+    else
+      raise """
+      You configuration is invalid, make sure you are using the correct options for this task.
+
+      #{format_errors(errors, " * ")}\
+      """
+    end
+  end
+
+  defp authenticate_cli!(config) do
+    log_debug("Authenticating CLI...")
+
+    case Teams.fetch_cli_session(config) do
+      {:ok, team} -> team
+      {:error, error} -> raise error
+      {:transport_error, error} -> raise error
+    end
+  end
+
+  defp deploy_to_teams(team, config) do
+    notebook_paths = list_notebooks!(config.path)
+    log_info("Deploying notebooks:")
+
+    for path <- notebook_paths do
+      log_info(" * Preparing to deploy notebook #{Path.basename(path)}")
+      files_dir = Livebook.FileSystem.File.local(path)
+
+      with {:ok, content} <- File.read(path),
+           {:ok, app_deployment} <- prepare_app_deployment(path, content, files_dir) do
+        case Livebook.Teams.deploy_app_from_cli(team, app_deployment, config.deployment_group) do
+          {:ok, url} ->
+            print_text([:green, "  * #{app_deployment.title} deployed successfully. (#{url})"])
+
+          {:error, errors} ->
+            print_text([:red, "  * #{app_deployment.title} failed to deployed."])
+            errors = normalize_errors(errors)
+
+            raise """
+            #{format_errors(errors, "  * ")}
+
+            #{Teams.Requests.error_message()}\
+            """
+
+          {:transport_error, reason} ->
+            print_text([:red, "  * #{app_deployment.title} failed to deployed."])
+            raise reason
+        end
+      end
+    end
+
+    :ok
+  end
+
+  defp list_notebooks!(path) do
+    log_debug("Listing notebooks from: #{path}")
+
+    files =
+      if File.dir?(path) do
+        path
+        |> File.ls!()
+        |> Enum.map(&Path.join(path, &1))
+        |> Enum.reject(&File.dir?/1)
+        |> Enum.filter(&String.ends_with?(&1, ".livemd"))
```

Comment:
```suggestion
        path
        |> Path.join("*.livemd")
        |> Path.wildcard()
        |> Enum.filter(&File.regular?/1)
```

However, should we deploy only the direct files in a directory or all files recursively? Another option is to not support directories and instead allow a list of files to be given. This way the user can write this:

    livebook deploy path/*.livemd

The shell will expand `*.livemd` into multiple entries and we just deploy all of them. WDYT?

Reply:
FTR for `LIVEBOOK_APPS_PATH` we do `path/**/*.livemd`:

https://github.com/livebook-dev/livebook/blob/6806ef8ec4ffb96c0dd4051775138f82fefa30ae/lib/livebook/apps.ex#L244-L247

But passing multiple files sounds good to me.